src/main/java/org/wso2/carbon/google/ads/connector/ClientCredentialsAccessTokenHandler.java

```diff
@@ -0,0 +1,199 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class ClientCredentialsAccessTokenHandler extends AbstractConnector {
+
+    private static final Log log = LogFactory.getLog(ClientCredentialsAccessTokenHandler.class);
+    private static final JsonParser parser = new JsonParser();
+    private static final String ERROR_MESSAGE = Constants.GENERAL_ERROR_MSG + "\"clientId\", \"clientSecret\"," +
+            " \"tokenEndpoint\", \"refreshToken\", \"developerToken\" parameters are mandatory.";
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+
+        String connectionName = (String) ConnectorUtils.
+                lookupTemplateParamater(messageContext, Constants.CONNECTION_NAME);
+
+        String base = (String) getParameter(messageContext, Constants.BASE);
+        if (StringUtils.endsWith(base, "/")) {
+            base = StringUtils.removeEnd(base, "/");
+        }
+        messageContext.setProperty(Constants.PROPERTY_BASE, base);
+
+        String clientId = (String) getParameter(messageContext, Constants.CLIENT_ID);
+        String clientSecret = (String) getParameter(messageContext, Constants.CLIENT_SECRET);
+        String tokenEndpoint = (String) getParameter(messageContext, Constants.TOKEN_ENDPOINT);
+        String refreshToken = (String) getParameter(messageContext, Constants.REFRESH_TOKEN);
+        String developerToken = (String) getParameter(messageContext, Constants.DEVELOPER_TOKEN);
+
+        if (StringUtils.isBlank(clientId) || StringUtils.isBlank(clientSecret)
+                || StringUtils.isBlank(tokenEndpoint) || StringUtils.isBlank(refreshToken) || StringUtils.isBlank(developerToken)){
+            Utils.setErrorPropertiesToMessage(messageContext, Constants.ErrorCodes.INVALID_CONFIG, ERROR_MESSAGE);
+            handleException(ERROR_MESSAGE, messageContext);
+        }
+
+        Map<String, String> payloadParametersMap = new HashMap<>();
+        payloadParametersMap.put(Constants.OAuth2.REFRESH_TOKEN, refreshToken);
+        payloadParametersMap.put(Constants.OAuth2.CLIENT_ID, clientId);
+        payloadParametersMap.put(Constants.OAuth2.CLIENT_SECRET, clientSecret);
+
+        Token token = TokenManager.getToken(getTokenKey(connectionName, tokenEndpoint, developerToken, payloadParametersMap));
```

Comment:
Move `getTokenKey()` to var and reuse to get and remove.

Reply:
Fixed with [fa7b308](https://github.com/wso2-extensions/esb-connector-googleads/pull/1/commits/fa7b308f94a8a150e1d96097a6df04f4378772c0)